g2p_programs/models/managers/cycle_manager.py

```diff
@@ -233,14 +233,55 @@ def mark_ended(self, cycle):
     def mark_cancelled(self, cycle):
         cycle.update({"state": constants.STATE_CANCELLED})
 
-    def validate_entitlements(self, cycle, cycle_memberships):
+    def validate_entitlements(self, cycle, entitlement_ids):
         # TODO: call the program's entitlement manager and validate the entitlements
         # TODO: Use a Job attached to the cycle
         # TODO: Implement validation workflow
         for rec in self:
-            rec.program_id.get_manager(
-                constants.MANAGER_ENTITLEMENT
-            ).validate_entitlements(cycle_memberships)
+            rec._ensure_can_edit_cycle(cycle)
+            rec.program_id.get_manager(constants.MANAGER_ENTITLEMENT)
+            if len(entitlement_ids) < 200:
+                self._validate_entitlements(entitlement_ids)
+            else:
+                self._validate_entitlements_async(cycle, entitlement_ids)
+
+    def _validate_entitlements_async(self, cycle, entitlement_ids):
+        _logger.debug("Validate entitlement asynchronously")
+        cycle.message_post(
+            body=_("Validation for %s entitlements started.", len(entitlement_ids))
+        )
+        cycle.write(
+            {
+                "locked": True,
+                "locked_reason": _("Validate entitlement for beneficiaries."),
+            }
+        )
+
+        jobs = []
+        max_jobs_per_batch = 100
+        ctr = 0
+        entitlements = []
+        max_rec = len(entitlement_ids)
+        for ctr_entitlements, entitlement in enumerate(entitlement_ids, 1):
+            ctr += 1
```

Comment:
You don't need a new variable for this you could use `ctr_entitlements % max_jobs_per_batch`

Reply:
Done @jeremi